src/main/java/model/FruitTransaction.java

```diff
@@ -0,0 +1,53 @@
+package model;
+
+public class FruitTransaction {
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(String symbol, String fruit, int quantity) {
+        this.operation = convertStringIntoOperation(symbol);
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    private Operation convertStringIntoOperation(String symbol) {
+        return switch (symbol) {
+            case "b" -> Operation.BALANCE;
+            case "s" -> Operation.SUPPLY;
+            case "p" -> Operation.PURCHASE;
+            case "r" -> Operation.RETURN;
+            default -> throw new IllegalArgumentException(
+                    "There is no Enum element, corresponding to the argument " + symbol);
+        };
```

Comment:
Avoid using switch-case constructions. It is recommended to use the Strategy pattern instead. You can create a map where the key is the string symbol and the value is the corresponding enum. This way you can just get the necessary enum from the map by the symbol.